src/config.js

```diff
@@ -1,39 +1,109 @@
 import collector from './collector';
+import packageConfig from './packageConfig';
+
 const { getHostname, getCollectorPath } = collector;
 
 module.exports = function setConfig(configObject) {
   const defaults = {
+    clientId: '',
+    debug: false,
     host: getHostname(),
-    path: getCollectorPath(),
-    clientId: process.env.IOPIPE_TOKEN || process.env.IOPIPE_CLIENTID || '',
-    debug: process.env.IOPIPE_DEBUG || false,
-    networkTimeout: 5000,
-    timeoutWindow: Number(process.env.IOPIPE_TIMEOUT_WINDOW) || 150,
     installMethod: 'manual',
-    plugins: []
+    networkTimeout: 5000,
+    path: getCollectorPath(),
+    plugins: [],
+    timeoutWindow: 150
   };
 
   const config = Object.assign({}, defaults);
+
+  const packageConf = packageConfig.getConfig();
+
+  // Override default config with package config variables
+  if (packageConf) {
```

Comment:
This is really big (and thus hard for me to reason about) -- I'd like to see it refactored into smaller bits. Or in a later comment, I suggested we build the config, rather than use a default override situation. With this much work happening after `Object.assign`, the defaults pattern doesn't look like it's working